src/i18nDefinitionProvider.ts

```diff
@@ -35,72 +25,24 @@ export class I18nDefinitionProvider implements DefinitionProvider {
-                return null;
-            }
-            return new Location(file, new Range(document.positionAt(range[0]), document.positionAt(range[1])));
-        });
-    }
-
-    findKeyValueRangeInYAML(yaml: string, absoluteKey: string, locale: string): number[] {
-        let yamlDocument: YAMLDocument = null;
-        try {
-            yamlDocument = YAML.parseDocument(yaml);
-        } catch (error) {
-            logger.error('could not parse yaml document', { error })
-            return null;
-        }
-        return this.findKeyValueRangeInYamlDocument(yamlDocument, absoluteKey, locale);
-    }
-
-    findKeyValueRangeInYamlDocument(yamlDocument: YAMLDocument, absoluteKey: string, locale: string): number[] {
-        logger.debug('findKeyValueRangeInYamlDocument', { absoluteKey, locale });
-
-        const keyParts: string[] = absoluteKey.split('.').filter(key => key.length > 0);
-
-        let yamlPairs = yamlDocument.contents.items;
-        if (!yamlPairs) {
-            logger.warn('yamlDocument does not have any items');
-            return null;
-        }
-
-        keyParts.unshift(locale);
-
-        for (let i = 0; i < keyParts.length; i++) {
-            const keyPart = keyParts[i];
-            const flatKey: string = keyParts.slice(i).join('.');
-            let yamlPair = yamlPairs.find(item => item.stringKey === flatKey);
-            if (!yamlPair) {
-                yamlPair = yamlPairs.find(item => item.stringKey === keyPart);
-                if (!yamlPair) {
-                    logger.debug('key could not be located in yaml document');
+            if (!yamlDocument) {
+                try {
+                    yamlDocument = YAMLDocument.parse(document.getText());
+                } catch (error) {
+                    logger.error('could not parse yaml document', { error })
                     return null;
                 }
             }
-            let value = yamlPair.value;
-            logger.debug('current value:', value, 'typeof value', typeof value);
-
-            if (typeof value !== 'object') {
-                logger.debug('unknown value object (not an object)', { value });
-                return null;
-            }
-
-            if (['PLAIN', 'QUOTE_DOUBLE', 'QUOTE_SINGLE', 'BLOCK_FOLDED', 'BLOCK_LITERAL'].indexOf(value.type) >= 0) {
-                logger.debug('findKeyValueRangeInYamlDocument', { value });
-                return value.range;
-            } else if (value.items) {
-                yamlPairs = value.items;
-            } else {
-                logger.debug('unknown value object (complex type with no child items)', { ...value });
+            const range: number[] = yamlDocument.findKeyValueRange(absoluteKey, locale);
+            if (!range) {
                 return null;
             }
-        }
-        return null;
+            return new Location(file, new Range(document.positionAt(range[0]), document.positionAt(range[1])));
+        });
     }
-}
```

Comment:
I moved all this stuff behind a wrapper class with my new code. Didn't want to let the Document stuff from the YAML package to bleed everywhere, so made sense to hide behind an interface that deals with all the internals in one place.